descartes_light/solvers/include/descartes_light/solvers/bgl/impl/bgl_dijkstra_solver.hpp

```diff
@@ -26,14 +28,16 @@ SearchResult<FloatType> BGLDijkstraSVSESolver<FloatType>::search()
   auto color_prop_map = boost::get(&Vertex<FloatType>::color, graph_);
   auto distance_prop_map = boost::get(&Vertex<FloatType>::distance, graph_);
 
-  predecessor_map_.clear();
-  boost::associative_property_map<std::map<VertexDesc<FloatType>, VertexDesc<FloatType>>> predecessor_prop_map(
-      predecessor_map_);
+  typedef typename boost::property_map<BGLGraph<FloatType>, boost::vertex_index_t>::type IndexMap;
+  typedef boost::iterator_property_map<typename std::vector<VertexDesc<FloatType>>::iterator, IndexMap> VertMap;
```

Comment:
I suspect this could be condensed further

Reply:
I would rename the typedef to `PredecessorMap` since it's mapping predecessors not just vertices in general